### PR TITLE
UAR-2184: Fixing logic in canGenerateRequestsInFuture in ProposalDeve…

### PR DESCRIPTION
**Pull request by nataliac06**
…lopmentActionsAction that determines if the user sees the question

### CODE REVIEW

File: src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java [hunk 1]
```diff
@@ -291,11 +292,13 @@ private boolean canGenerateRequestsInFuture( WorkflowDocument workflowDoc, Strin
 
 		org.kuali.rice.krad.workflow.service.WorkflowDocumentService workflowDocumentService = KRADServiceLocatorWeb.getService( "workflowDocumentService" );
 		String currentRouteNodeNames = workflowDocumentService.getCurrentRouteNodeNames( workflowDoc );
-
-		return ( hasAskedToNotReceiveFutureRequests( workflowDoc, principalId ) && canGenerateMultipleApprovalRequests( reportCriteriaBuilder.build(), principalId, currentRouteNodeNames ) );
+		boolean result = hasAskedToReceiveFutureRequests( workflowDoc, principalId ) && canGenerateMultipleApprovalRequests( reportCriteriaBuilder.build(), principalId, currentRouteNodeNames );
+		LOG.info("canGenerateRequestsInFuture currentRouteNodeNames="+currentRouteNodeNames+" principalId="+principalId+ " result="+result);
```
Comment (Perrinek) [inline src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java]: Should this be LOG.debug instead? Any concerns about the number of log entries caused by the LOG.info?

File: src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java [hunk 2]
```diff
@@ -308,37 +311,51 @@ private boolean hasAskedToNotReceiveFutureRequests( WorkflowDocument workflowDoc
 				if ( variableKey.startsWith( KewApiConstants.RECEIVE_FUTURE_REQUESTS_BRANCH_STATE_KEY )
 						&& variableValue.toUpperCase().equals( KewApiConstants.RECEIVE_FUTURE_REQUESTS_BRANCH_STATE_VALUE )
 						&& variableKey.contains( principalId ) ) {
+					LOG.info("hasAskedToReceiveFutureRequests: receiveFutureRequests = true");
```
Comment (Perrinek) [inline src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java]: Should this be LOG.debug instead? Any concerns about the number of log entries caused by the LOG.info?

File: src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java [hunk 3]
```diff
@@ -308,37 +311,51 @@ private boolean hasAskedToNotReceiveFutureRequests( WorkflowDocument workflowDoc
...
 					receiveFutureRequests = true;
-					break;
 				}
 				else if ( variableKey.startsWith( KewApiConstants.RECEIVE_FUTURE_REQUESTS_BRANCH_STATE_KEY )
 						&& variableValue.toUpperCase().equals( KewApiConstants.DONT_RECEIVE_FUTURE_REQUESTS_BRANCH_STATE_VALUE )
 						&& variableKey.contains( principalId ) ) {
+					LOG.info("hasAskedToReceiveFutureRequests: doNotReceiveFutureRequests = true");
```
Comment (Perrinek) [inline src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java]: Should this be LOG.debug instead? Any concerns about the number of log entries caused by the LOG.info?

File: src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java [hunk 4]
```diff
@@ -308,37 +311,51 @@ private boolean hasAskedToNotReceiveFutureRequests( WorkflowDocument workflowDoc
...
 					doNotReceiveFutureRequests = true;
 					break;
 				}
 			}
 		}
 
-		return ( receiveFutureRequests == false && doNotReceiveFutureRequests == false );
+		//has to account for both cases, when the user first wants to receive further requests, but then later on decides he doesn't want to
+		LOG.info("hasAskedToReceiveFutureRequests: returning ="+ (!doNotReceiveFutureRequests || receiveFutureRequests));
```
Comment (Perrinek) [inline src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java]: Should this be LOG.debug instead? Any concerns about the number of log entries caused by the LOG.info?

File: src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java [hunk 5]
```diff
@@ -308,37 +311,51 @@ private boolean hasAskedToNotReceiveFutureRequests( WorkflowDocument workflowDoc
...
 		for ( ActionRequest actionRequest : results1.getActionRequests() ) {
-			// !actionRequest.isRoleRequest() && removed from condition for
-			if ( actionRequest.isPending() && actionRequest.getActionRequested().getCode().equalsIgnoreCase( KewApiConstants.ACTION_REQUEST_APPROVE_REQ ) &&
-					recipientMatchesUser( actionRequest, loggedInPrincipalId ) && !StringUtils.contains( currentRouteNodeNames, actionRequest.getNodeName() ) ) {
-				approvalRequestsCount += 1;
+			if ( actionRequest.isPending()
+			        && actionRequest.getActionRequested().getCode().equalsIgnoreCase( KewApiConstants.ACTION_REQUEST_APPROVE_REQ ) 
+			        && recipientMatchesUser( actionRequest, loggedInPrincipalId ) )	{
+				crtUserActiveActionRequestsCount++;
+				LOG.info("MultipleApprovalRequests: actionRequestId="+actionRequest.getId()+" actionRequestAnnotation="+actionRequest.getAnnotation()+" crtUserActiveActionRequests="+crtUserActiveActionRequestsCount);
```
Comment (Perrinek) [inline src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java]: Should this be LOG.debug instead? Any concerns about the number of log entries caused by the LOG.info?

File: src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java [hunk 6]
```diff
@@ -308,37 +311,51 @@ private boolean hasAskedToNotReceiveFutureRequests( WorkflowDocument workflowDoc
...
+				if ( crtUserActiveActionRequestsCount >1 ){
+					return true;
+				}
 			}
 		}
-
-		return ( approvalRequestsCount > 0 );
+		LOG.info("canGenerateMultipleApprovalRequests: crtUserActiveActionRequestsCount="+crtUserActiveActionRequestsCount +" Returning FALSE");
```
Comment (Perrinek) [inline src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java]: Should this be LOG.debug instead? Any concerns about the number of log entries caused by the LOG.info?

Comment (nataliac06) [reply src/main/java/org/kuali/kra/proposaldevelopment/web/struts/action/ProposalDevelopmentActionsAction.java]: I will change all the statements to Log.debug.

**Comment by Perrinek** [comment]
Looks good!

**Pull request merged**